### PR TITLE
Unit test submitBlock

### PR DESCRIPTION
**Pull request by fleupold**
More unit tests for submitBlock

### CODE REVIEW

File: test/plasma.js [hunk 1]
```diff
@@ -52,37 +54,45 @@ contract('Plasma', (accounts) => {
...
+      await plasma.startDepositExit(position, 0, oneETH, {from: depositor})
+      
+      after = (await plasma.exits.call(position))
       assert.equal(after[0], depositor);
     })
   })
 
-  describe('submitBlock: Trivial Tests', () => {
+  describe('submitBlock', () => {
     BlockType = {
```
Comment (josojo) [inline test/plasma.js]: Maybe we should put the BlockType into the utilities.js

**Review by josojo — APPROVED** [review]
I have never seen such a mock contract usage. Neat

File: test/plasma.js (hunk 1 above)
Comment (bh2smith) [reply test/plasma.js]: I like that!

File: contracts/Plasma.sol [hunk 2]
```diff
@@ -161,41 +161,39 @@ contract Plasma {
...
-        } else if (_blockType == BlockType.AuctionOutput) {  // AuctionOutput
+        } else if (_blockType == BlockType.AuctionOutput) {
             require(
                 prevBlockType == BlockType.AuctionResult,
                 "Auction Output must come immediately after Auction Result Block"
             );
         }
 
-        // THE REST!
+        // Create the block structure
```
Comment (bh2smith) [inline contracts/Plasma.sol:196]: When we split these contracts, this is where we should call
```
super.submitBlock(parameters);
```

**Review by bh2smith — APPROVED** [review]
All of these changes look really great. I don't know how you are so fast with all this stuff.

**Pull request merged**